internal/app/donor_store.go

```diff
@@ -335,6 +335,32 @@ func (s *donorStore) Delete(ctx context.Context) error {
 	return s.dynamoClient.DeleteKeys(ctx, keys)
 }
 
+func (s *donorStore) DeleteLink(ctx context.Context, shareCodeData actor.ShareCodeData) error {
+	data, err := page.SessionDataFromContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	if data.OrganisationID == "" {
+		return errors.New("donorStore.DeleteLink requires OrganisationID")
+	}
+
+	if data.OrganisationID != shareCodeData.SessionID {
```

Comment:
Slight overkill but as we had the data handy thought we may as well be sure the remover has the rights to do it.